### ISSUE TITLE
Panic while parsing RIB dump (called `Option::unwrap()` on a `None` value)

### ISSUE THREAD

**Issue by mbUSC** [bgpkit/bgpkit-parser#158]
I ran into this panic while parsing a rib dump from 2019:
```
thread 'main' panicked at /home/cs/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bgpkit-parser-0.10.5/src/parser/mrt/mrt_elem.rs:367:34:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/panicking.rs:72:14
   2: core::panicking::panic
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/panicking.rs:145:5
   3: core::option::unwrap_failed
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/option.rs:1985:5
   4: core::option::Option<T>::unwrap
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/option.rs:933:21
   5: bgpkit_parser::parser::mrt::mrt_elem::Elementor::record_to_elems
             at /home/cs/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bgpkit-parser-0.10.5/src/parser/mrt/mrt_elem.rs:362:40
   6: <bgpkit_parser::parser::iters::ElemIterator<R> as core::iter::traits::iterator::Iterator>::next
             at /home/cs/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bgpkit-parser-0.10.5/src/parser/iters.rs:166:45
   7: alloc::vec::Vec<T,A>::extend_desugared
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/alloc/src/vec/mod.rs:2988:35
   8: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/alloc/src/vec/spec_extend.rs:17:9
   9: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/alloc/src/vec/spec_from_iter_nested.rs:43:9
  10: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/alloc/src/vec/spec_from_iter.rs:33:9
  11: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/alloc/src/vec/mod.rs:2894:9
  12: core::iter::traits::iterator::Iterator::collect
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/iter/traits/iterator.rs:2003:9
  13: bgpkit_test::main
             at ./src/main.rs:17:21
  14: core::ops::function::FnOnce::call_once
             at /rustc/a165f1f65015b1bd4afd2ec50700aaacf2e0c485/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

This is the code to reproduce the problem:
```
use bgpkit_broker::BgpkitBroker;
use bgpkit_parser::{BgpElem, BgpkitParser};

fn main() {

    let broker = BgpkitBroker::new()
        .ts_start("2019-02-16T00:00:00+00:00")
        .ts_end("2019-02-16T00:00:00+00:00")
        .data_type("rib")
        .collector_id("route-views.sg");

    for item in &broker {
        let parser = BgpkitParser::new(&item.url)
            .unwrap();

        let elems = parser
            .into_elem_iter()
            .collect::<Vec<BgpElem>>();

        println!("{:?}", elems);
    }
}
```

**Comment by digizeph** [comment]
Thank you @mbUSC for the bug report!

I can confirm with the following command:
```
bgpkit-parser http://archive.routeviews.org/route-views.sg/bgpdata/2019.02/RIBS/rib.20190216.0000.bz2 > /dev/null

thread 'main' panicked at src/parser/mrt/mrt_elem.rs:367:34:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The relevant code snippet that cased the issue is: 
https://github.com/bgpkit/bgpkit-parser/blob/56c4d096f39c06711dd765e4cef9eb94173484a0/src/parser/mrt/mrt_elem.rs#L362-L367

At a glance, this seems to indicate that there is a malformed RIB entry with a peer ID not existing on the initial peer index table of the RIB dump file. I've never seen this happened before but I will get rid of the `unwrap` and trigger warnings instead. Stay tuned for the bug fix release coming soon!

**Comment by digizeph** [comment]
Cross-checking with other MRT readers to confirm the theory that the dump data is corrupted:

`bgpdump`
```
bgpdump -m rib.20190216.0000.bz2 > /dev/null
2024-03-30 11:07:40 [info] logging to syslog
Assertion failed: (e->peer_index < entry->dump->table_dump_v2_peer_index_table->peer_count), function process_mrtd_table_dump_v2_ipv4_unicast, file bgpdump_lib.c, line 637.
[1]    94046 abort      bgpdump -m rib.20190216.0000.bz2 > /dev/null
```

`bgpreader`
```
bgpreader -d singlefile -o rib-file=rib.20190216.0000.bz2 > /dev/null
WARN: No time window specified, defaulting to all available data
WARN: INVALID_MSG: Path attribute (type 10) has length 75, but only 19 bytes remain. (parsebgp_bgp_update.c:555)
WARN: INVALID_MSG: Path attribute (type 10) has length 75, but only 19 bytes remain. (parsebgp_bgp_update.c:555)
```

**Comment by digizeph** [comment]
This is confirmed as a malformed RIB dump entries causing panics. I have released [`v0.10.6`](https://github.com/bgpkit/bgpkit-parser/releases/tag/v0.10.6) and should avoid panics while processing similar malformed messages.

Let me know if this addresses your issue. Thanks!

**Comment by mbUSC** [comment]
Yes, that fixed it -- thank you!
